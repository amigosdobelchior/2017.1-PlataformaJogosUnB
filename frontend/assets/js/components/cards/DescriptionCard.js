import React from "react";
import PropTypes from 'prop-types';
import {Card} from "semantic-ui-react";

export default class DescriptionCard extends React.Component {
    
    getAwards(awards){
        console.log(awards)
        if (awards.length > 0) {
            return (<Card.Content header='Prêmios' description={awards}/>)
        } else {
            return null
        }
    }

    render () {
        const awards = (this.props.awards.map((award, i) =>
                        <p key={i}>
                            {this.props.getFields('Nome do Prêmio: ', award.name, ' - ')}
                            {this.props.getFields('Ano: ', award.year, ' - ')}
                            {this.props.getFields('Colocação: ', award.place, '')}
                        </p>
                    ));

        return (
            <Card fluid>
              <Card.Content header='Descrição' description={this.props.description} />
                <Card fluid>
                {this.getAwards(awards)}
              </Card>
            </Card>
        );
    }
}

DescriptionCard.propTypes = {
    description: PropTypes.string.isRequired,
    awards: PropTypes.array.isRequired,
    getFields: PropTypes.func.isRequired,
}
