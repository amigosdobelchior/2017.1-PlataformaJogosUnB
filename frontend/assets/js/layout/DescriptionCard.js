import React from 'react';
import { Card } from 'semantic-ui-react';

export default class DescriptionCard extends React.Component{
  getVersion(){
    if (this.props.version != null){
      return(<p><strong>Versão: </strong>{this.props.version}</p>);
    }else{
      return (null);
    }
  };
  
  render(){
    return(
      <Card fluid={true}>
        <Card.Content>
          <Card.Header>
            Sobre o jogo
          </Card.Header>
          <Card.Description>   
            { this.getVersion() }
            <p><strong>Repositório Oficial: </strong>{this.props.official_repository}</p>
            <p><strong>Ano de lançamento: </strong>{this.props.launch_year}</p>
            <p><strong>Gênero: </strong>{this.props.name}</p>
            <p><strong>Developers: </strong> {this.props.developers.map((developer) => {return developer.name} )}</p>
          </Card.Description>
        </Card.Content>
      </Card>
    ); 
  }
}
