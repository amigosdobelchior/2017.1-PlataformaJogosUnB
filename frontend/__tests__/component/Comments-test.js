import React from 'react';
import DescriptionCard from '../../assets/js/components/Comments';
import FacebookProvider, { Comments } from 'react-facebook';
import renderer from 'react-test-renderer';
var expect = required('expect');

test('Test render Comments', () => {
  const component = renderer.create(
		<div>
      <FacebookProvider appId="1850394608544081">
        <Comments href="unbgames.lappis.rocks/games/1" colorScheme={'dark'} width='100%' />
      </FacebookProvider>
    </div>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
