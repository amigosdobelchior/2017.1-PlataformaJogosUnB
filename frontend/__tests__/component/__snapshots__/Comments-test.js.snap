// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Comments 1`] = `
<div>
  <div
    className={undefined}
  >
    <div
      className="fb-comments"
      data-colorscheme="dark"
      data-href="unbgames.lappis.rocks/games/1"
      data-numposts={10}
      data-order-by="social"
      data-skin="dark"
      data-width="100%"
    />
  </div>
</div>
`;
